block-library/custom-query-loop/block/edit.js

```diff
@@ -0,0 +1,1145 @@
+	 *
+	 * @description grabs post suggestions from provided post types
+	 *
+	 * @return {Array} array of suggestions
+	 */
+	const postInSuggestions = () => {
+		const suggestions = [];
+
+		if ( postTypes !== null && postTypes.length > 0 ) {
+			for ( let i = 0; i < postTypes.length; i++ ) {
+				if ( postInPosts[ postTypes[ i ].slug ] !== null ) {
+					for (
+						let j = 0;
+						j < postInPosts[ postTypes[ i ].slug ].length;
+						j++
+					) {
+						suggestions.push(
+							postInPosts[ postTypes[ i ].slug ][ j ].title.rendered
+						);
+					}
+				}
+			}
+		}
+
+		return suggestions;
+	};
+
+	/**
+	 * @function getTaxonomyTermSuggestions
+	 *
+	 * @description grabs term suggestions from provided taxonomy
+	 *
+	 * @param {*} taxonomy
+	 *
+	 * @return {Array} returns array of related term names
+	 */
+	const getTaxonomyTermSuggestions = ( taxonomy ) => {
+		return taxonomyTerms !== null &&
+			taxonomyTerms[ taxonomy ] !== undefined &&
+			taxonomyTerms[ taxonomy ] !== null
+			? taxonomyTerms[ taxonomy ].map( ( term ) => {
+					return term.name;
+			  } )
+			: [];
+	};
+
+	/**
+	 * @function postTypeSuggestions
+	 *
+	 * @description used to populate the post types FormTokenField suggestions
+	 *
+	 * @return {Array} returns array of post type names
+	 */
+	const postTypeSuggestions = () => {
+		return postTypesList !== null
+			? postTypesList.map( ( postType ) => {
+					return postType.name;
+			  } )
+			: [];
+	};
+
+	return (
+		<div { ...blockProps }>
+			<ServerSideRender
+				block="tribe/custom-query-loop"
+				attributes={ props.attributes }
+			/>
+			{ isSelected && (
+				<InspectorControls>
+					<PanelBody title={ __( 'Block Settings', 'tribe' ) }>
+						<SelectControl
+							__nextHasNoMarginBottom={ true }
+							__next40pxDefaultSize={ true }
+							label={ __( 'Card Component', 'tribe' ) }
+							help={ __(
+								'The card component to be used in the query loop',
+								'tribe'
+							) }
+							value={ component }
+							options={ [
+								{ value: 'post', label: 'Post' },
+								{ value: 'search', label: 'Search' },
+							] }
+							onChange={ ( value ) => {
+								setAttributes( {
+									component: value,
+								} );
+							} }
+						/>
+						<RangeControl
+							__nextHasNoMarginBottom={ true }
+							__next40pxDefaultSize={ true }
+							label={ __( 'Number of Columns', 'tribe' ) }
+							help={ __(
+								'The number of columns to display the selected components in.'
+							) }
+							value={ columns }
+							initialPosition={ columns }
+							max={ 4 }
+							min={ 1 }
+							onChange={ ( value ) => {
+								setAttributes( {
+									columns: value,
+								} );
+							} }
+						/>
+						<FormTokenField
+							__next40pxDefaultSize={ true }
+							label={ __( 'Post Types', 'tribe' ) }
+							value={ postTypes }
+							suggestions={ postTypeSuggestions() }
+							onChange={ ( tokens ) => {
+								setPostTypes( tokens );
+							} }
```

Comment:
I am finding these form token fields a little confusing to use. Maybe we can add more helper text or change the control type? It just wasn't clear to me that typing would bring up a list of available post types.